File: module_music/src/main/java/com/yyxnb/module_music/db/MusicLocalDao.java
package com.yyxnb.module_music.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;

import com.yyxnb.common_base.db.BaseDao;
import com.yyxnb.module_music.bean.MusicLocalBean;

import java.util.List;

/**
 * 本地音乐
 */
@Dao
public interface MusicLocalDao extends BaseDao<MusicLocalBean> {

    @Query("SELECT * FROM local")
    LiveData<List<MusicLocalBean>> liveMusicList();

    @Query("SELECT * FROM local")
    List<MusicLocalBean> getMusicList();

    //删全部
    @Query("DELETE FROM local")
    void deleteAll();

//    @Insert(onConflict = OnConflictStrategy.REPLACE)
//    void insertItems(ArrayList<IAudio> mLists);
}

File: module_music/src/main/java/com/yyxnb/module_music/db/MusicLocalDao.kt
package com.yyxnb.module_music.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.yyxnb.common_base.db.BaseDao
import com.yyxnb.module_music.bean.MusicLocalBean

/**
 * 本地音乐
 */
@Dao
interface MusicLocalDao : BaseDao<MusicLocalBean> {
    @Query("SELECT * FROM local")
    fun liveMusicList(): LiveData<List<MusicLocalBean>>

    @get:Query("SELECT * FROM local")
    val musicList: List<MusicLocalBean>

    //删全部
    @Query("DELETE FROM local")
    fun deleteAll() //    @Insert(onConflict = OnConflictStrategy.REPLACE)
    //    void insertItems(ArrayList<IAudio> mLists);
}